doc/GETTING_STARTED.md

```diff
@@ -29,6 +30,14 @@ npm install libp2p
 
 If you're new to libp2p, we recommend configuring your node in stages, as this can make troubleshooting configuration issues much easier. In this guide, we'll do just that. If you're more experienced with libp2p, you may wish to jump to the [Configuration readme](./CONFIGURATION.md).
 
+## ESM
+
+Since `libp2p@0.37.0` is that the module is now [ESM-only](https://gist.github.com/sindresorhus/a39789f98801d908bbc7ff3ecc99d99c).
+
+ESM is the module system for JavaScript, it allows us to structure our code in separate files without polluting a global namespace.
+
+Other systems have tried to fill this gap, notably CommonJS, AMD, RequireJS and others, but ESM is [the official standard format](https://tc39.es/ecma262/#sec-modules) to package JavaScript code for reuse.
+
```

Comment:
I think it’s worth adding a section with the minimal typescript config for generating ESM here too

Reply:
Well someone could use ESM modules without Typescript but I suppose we suggest that  the `type` parameter be set to `module` as mentioned in the [node docs](https://nodejs.org/api/esm.html#enabling)

Reply:
That is of course true, but by default TypeScript outputs CJS which then explodes at runtime.

People are then confused because since TypeScript uses `import` it looks like they are authoring ESM when in fact they are not.

A cursory search through the closed issues in js-IPFS around the time we converted to ESM will reveal that this is a very common mistake.